### PR TITLE
AI boilerplate as new MASTER

### PR DESCRIPTION
**Pull request by tumido**
Update `master` with the AI boilerplate

This PR is extracting the server side boilerplate @AparnaKarve wrote for the volume type validation.

@AparnaKarve, @jameswnl can you please check if I don't miss anything?

### CODE REVIEW

**Review by AparnaKarve — APPROVED** [review]
Looks great!

Will wait for @jameswnl 's input.

File: workers.py [hunk 1]
```diff
@@ -0,0 +1,94 @@
+import logging
+from threading import Thread, current_thread
+
+import requests
+
+logger = logging.getLogger()
```
Comment (jameswnl) [inline workers.py]: 'logger' capital global constant?

File: workers.py [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+MAX_RETRIES = 3
+
+
+def _retryable(method: str, *args, **kwargs) -> requests.Response:
```
Comment (jameswnl) [inline workers.py:10]: seeing this in several repo. shall we create a Swiss knife package?

Comment (tumido) [reply workers.py:10]: No need for that now... It's gonna get removed or consolidated into a package with the rearch... No need for a standalone package for it, it'll be able to live as a util in a generic lib for workflow tasks.

**Review by jameswnl — APPROVED** [review]

**Review by jameswnl — APPROVED** [review]

**Review by AparnaKarve — APPROVED** [review]
LGTM. Merging...

**Pull request merged**